tron/serialize/runstate/dynamodb_state_store.py

```diff
@@ -152,19 +181,20 @@ def _consume_save_queue(self):
         for _ in range(qlen):
             try:
                 with self.save_lock:
-                    key, val = self.save_queue.popitem(last=False)
+                    key, (pickled_val, json_val) = self.save_queue.popitem(last=False)
+                log.debug(f"Processing save for {key} with a value of {pickled_val}")
                 # Remove all previous data with the same partition key
                 # TODO: only remove excess partitions if new data has fewer
                 self._delete_item(key)
-                if val is not None:
-                    self[key] = pickle.dumps(val)
+                if pickled_val is not None:
+                    self.__setitem__(key, pickle.dumps(pickled_val), json_val)
```

Comment:
just curious: how come we need `__setitem__` now?

Reply:
ah, i see - i think we probably want to keep the existing signature since `__setitem__` is an established "dunder" method in python

what we'd probably need to do here instead is to type that the value for `__setitem__` is going to be a tuple :)